OrbitLinuxTracing/TracerThread.h

```diff
@@ -114,6 +113,28 @@ class TracerThread {
   };
 
   EventStats stats_;
+
+  template <typename SamplePerfEventT>
+  static std::unique_ptr<SamplePerfEventT> ConsumeSamplePerfEvent(
```

Comment:
Shouldn't this be a member of PerfEventRingBuffer?  If not, why not?

Reply:
I was really unsure where to put this. The reason why it's not in PerfEventRingBuffer is because I wanted to keep it independent of PerfEvent, as the ring buffer is conceptually more lower level while PerfEvents are our own events.

Reply:
The ring buffer is already called PerfEventRingBuffer, so consuming PerfEvents in there seems exactly what it's intended for, no? (If not, it should be renamed and be a generic ring buffer. I don't think we would gain anything from that though.) This method follows the same pattern as the other consume methods, so I think it belongs in PerfEventRingBuffer.

Reply:
Yeah, the naming is unfortunate, but PerfEventRingBuffer doesn't read PerfEvents, it reads _perf_event_ _open records (it has the notion of header, for example). That's why it's lower level but still not a generic ring buffer. We could have this method read a dynamically_sized_perf_event_stack_sample but then we should move that struct out of PerfEvent.h into PerfEventRecords.h. Still, dynamically_sized_perf_event_stack_sample is not a real perf_event_open record so that's why it wasn't done in the first place.

Reply:
I'm not sure yet I understand which value you see in not having PerfEventRingBuffer also read PerfEvents. I don't think we need to keep this conceptually distinct. In how the consume method here is used, it would fit the class.

Reply:
My suggestion here would be to increase the scope of what PerfEventRingBuffer can do: It's a class to read from a ring buffer and convert into the data we need for the next step. Then this, and other similar methods fit naturally into the class (we will need this to read GPU tracing events from the ring buffers as well).

Reply:
I move this function into the new PerfEventReaders.h.